test/metabase/query_processor/util/add_alias_info_test.clj

```diff
@@ -647,23 +647,24 @@
   (testing "[[add/alias-from-join]] should match Fields in the Join source query even if they have temporal units"
     (mt/dataset sample-dataset
       (mt/with-everything-store
-        (is (= {:field-name              "CREATED_AT"
-                :join-is-this-level?     "Q2"
-                :alias-from-join         "Products__CREATED_AT"
-                :alias-from-source-query nil}
-               (#'add/expensive-field-info
-                (mt/$ids nil
-                  {:source-table $$reviews
-                   :joins        [{:source-query {:source-table $$reviews
-                                                  :breakout     [[:field %products.created_at
-                                                                  {::add/desired-alias "Products__CREATED_AT"
-                                                                   ::add/position      0
-                                                                   ::add/source-alias  "CREATED_AT"
-                                                                   ::add/source-table  "Products"
-                                                                   :join-alias         "Products"
-                                                                   :temporal-unit      :month}]]}
-                                   :alias        "Q2"}]})
-                [:field (mt/id :products :created_at) {:join-alias "Q2"}])))))))
+        (mt/with-driver :h2
```

Comment:
Something to keep in mind:

_technically_ you're not supposed to be doing `with-driver` inside of `mt/dataset` because `mt/dataset` binds the current test DB e.g. `(mt/db)` or `(mt/id ...)` to `sample-dataset` for the current driver, so changing the driver would mean you have the wrong driver's DB bound. So `with-driver` is supposed to happen around `dataset` rather than vice versa.

`mt/dataset` will throw an Exception if you **actually** change the driver, but if you use it with no driver bound it will assume `:h2` which means you're not actually changing the driver for the current DB.

At some point I might write a linter to catch this, otherwise I can't expect anyone to know that rule